lib/spack/spack/binary_distribution.py

```diff
@@ -1377,7 +1377,7 @@ def _build_tarball_in_stage_dir(
     return None
 
 
-def nodes_to_be_packaged(specs, root=True, dependencies=True):
+def specs_to_be_packaged(specs, root=True, dependencies=True) -> List[Spec]:
```

Comment:
Also here we can move the type hints from the docstring to the function signature.